### PR TITLE
[ISSUE-446] fake attach events

### PR DESCRIPTION
**Pull request by yurkov-anton**
### Issue #446

Add event FakeAttachEnabled, FakeAttachDisabled. Generated in NodeStageVolume and NodeUnstageVolume.

## PR checklist
- [X] Add link to the issue
- [ ] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

### CODE REVIEW

File: pkg/eventing/eventing.go [hunk 1]
```diff
@@ -45,4 +45,7 @@ const (
 	DriveHasData              = "DriveHasData"
 	DriveClean                = "DriveClean"
 	DriveHealthOverridden     = "DriveHealthOverridden"
+
+	FakeAttachEnabled  = "FakeAttachEnabled"
```
Comment (mishoyama) [inline pkg/eventing/eventing.go:49]: Please rename to `FakeAttachInvolved` and `FakeAttachCleared`

File: docs/fake-attach.md [hunk 2]
```diff
@@ -29,6 +29,8 @@ This feature will require application Operator to put specific annotation on PVC
 
 If fake-attach volume is successfully created, CSI Volume CR will be annotated with `fake-attach:yes` and `Operational Status` will equal to `MISSING`. The pod must be restarted after `Operational Status` returns to `OPERATIVE`.
 
+CSI must generate event to notify user about that Fake-Attach is enabled for this Volume.
```
Comment (mishoyama) [inline docs/fake-attach.md:32]: Please add details about events and when they are generated.

File: pkg/node/node.go [hunk 3]
```diff
@@ -224,6 +225,8 @@ func (s *CSINodeService) NodeStageVolume(ctx context.Context, req *csi.NodeStage
 	if isFakeAttach {
 		volumeCR.Annotations["fake-attach"] = "yes"
 		ll.Warningf("Adding fake-attach annotation to the volume with ID %s", volumeID)
+		s.VolumeManager.recorder.Eventf(volumeCR, eventing.NormalType, eventing.FakeAttachEnabled,
+			"Fake-attach enable for volume with ID %s", volumeID)
```
Comment (mishoyama) [inline pkg/node/node.go:229]: _Fake-attach involved for volume with ID_

File: pkg/node/node.go [hunk 4]
```diff
@@ -296,6 +299,8 @@ func (s *CSINodeService) NodeUnstageVolume(ctx context.Context, req *csi.NodeUns
 	if volumeCR.Annotations[fakeAttachVolumeAnnotation] == fakeAttachVolumeKey {
 		ll.Warningf("Removing fake-attach annotation for volume %s", volumeID)
 		delete(volumeCR.Annotations, fakeAttachVolumeAnnotation)
+		s.VolumeManager.recorder.Eventf(volumeCR, eventing.NormalType, eventing.FakeAttachDisabled,
+			"Fake-attach disable for volume with ID %s", volumeID)
```
Comment (mishoyama) [inline pkg/node/node.go:303]: _Fake-attach cleared for volume with ID_

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: pkg/node/node.go [hunk 5]
```diff
@@ -296,6 +299,8 @@ func (s *CSINodeService) NodeUnstageVolume(ctx context.Context, req *csi.NodeUns
 	if volumeCR.Annotations[fakeAttachVolumeAnnotation] == fakeAttachVolumeKey {
 		ll.Warningf("Removing fake-attach annotation for volume %s", volumeID)
 		delete(volumeCR.Annotations, fakeAttachVolumeAnnotation)
+		s.VolumeManager.recorder.Eventf(volumeCR, eventing.NormalType, eventing.FakeAttachDisabled,
```
Comment (safronovD) [inline pkg/node/node.go:302]: You should replace deleting the annotation and generating disable event to NodeStage. Or this event will appear each time when a related pod restarts

**Review by safronovD — APPROVED** [review]

**Review by mishoyama — APPROVED** [review]

**Pull request merged**